connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/ConnectAssignment.java

```diff
@@ -0,0 +1,255 @@
+            Collections.emptyList(), Collections.emptyList(), Collections.emptyList(), Collections.emptyList(), 0);
+
+    /**
+     * Create an assignment indicating responsibility for the given connector instances and task Ids.
+     *
+     * @param revokedConnectorIds list of connectors that the worker should stop running
+     * @param revokedTaskIds list of task IDs that the worker should stop running
+     * @param delay the scheduled delay after which the worker should rejoin the group
+     */
+    public ConnectAssignment(short version, short error, String leader, String leaderUrl, long configOffset,
+                             Collection<String> connectorIds, Collection<ConnectorTaskId> taskIds,
+                             Collection<String> revokedConnectorIds, Collection<ConnectorTaskId> revokedTaskIds,
+                             int delay) {
+        super(error, leader, leaderUrl, configOffset, connectorIds, taskIds);
+        this.version = version;
+        this.revokedConnectorIds = revokedConnectorIds;
+        this.revokedTaskIds = revokedTaskIds;
+        this.delay = delay;
+    }
+
+    /**
+     * Return the version of the connect protocol that this assignment belongs to.
+     *
+     * @return the connect protocol version of this assignment
+     */
+    public short version() {
+        return version;
+    }
+
+    /**
+     * Return the IDs of the connectors that are revoked by this assignment.
+     *
+     * @return the revoked connector IDs
+     */
+    public Collection<String> revokedConnectors() {
+        return revokedConnectorIds;
+    }
+
+    /**
+     * Return the IDs of the tasks that are revoked by this assignment.
+     *
+     * @return the revoked task IDs
+     */
+    public Collection<ConnectorTaskId> revokedTasks() {
+        return revokedTaskIds;
+    }
+
+    /**
+     * Return the delay for the rebalance that is scheduled by this assignment.
+     *
+     * @return the scheduled delay
+     */
+    public int delay() {
+        return delay;
+    }
+
+    /**
+     * Return an empty assignment.
+     *
+     * @return an empty assignment
+     */
+    public static ConnectAssignment empty() {
+        return EMPTY;
+    }
+
+    @Override
+    public String toString() {
+        return "Assignment{" +
+                "error=" + error() +
+                ", leader='" + leader() + '\'' +
+                ", leaderUrl='" + leaderUrl() + '\'' +
+                ", offset=" + offset() +
+                ", connectorIds=" + connectors() +
+                ", taskIds=" + tasks() +
+                ", revokedConnectorIds=" + revokedConnectorIds +
+                ", revokedTaskIds=" + revokedTaskIds +
+                ", delay=" + delay +
+                '}';
+    }
+
+    private Map<String, Collection<Integer>> revokedAsMap() {
+        if (revokedConnectorIds == null && revokedTaskIds == null) {
+            return null;
```

Comment:
see answer in the other comment about `asMap`. The two methods are not symmetric to this respect, because one field is required and the other one is optional.